server/database/judge.go

```diff
@@ -87,13 +88,13 @@ func UpdateJudges(db *mongo.Database, sc, judges []*models.Judge) error {
 
 // UpdateJudgesWithTx updates multiple judges in the database with a transaction
 func UpdateJudgesWithTx(db *mongo.Database, sc mongo.SessionContext, judges []*models.Judge) error {
+	models := make([]mongo.WriteModel, 0, len(judges))
 	for _, judge := range judges {
-		err := UpdateJudge(db, judge, sc)
-		if err != nil {
-			return err
-		}
+		models = append(models, mongo.NewUpdateOneModel().SetFilter(gin.H{"_id": judge.Id}).SetUpdate(gin.H{"$set": judge}))
 	}
-	return nil
+	opts := options.BulkWrite().SetOrdered(false)
+	_, err := db.Collection("judges").BulkWrite(context.Background(), models, opts)
```

Comment:
Instead of using `context.Background`, you should use the provided `sc` mongo session context.